Library/Homebrew/compilers.rb

```diff
@@ -111,6 +110,14 @@ def find_compiler
           version = versions.non_apple_gcc_version(name)
           yield Compiler.new(name, version) if version
         end
+      when :clang
+        # silently substitute clang to clang-omp when openmp is required.
+        version = if @formula.needs_compiler_feature?(:openmp)
+          versions.clang_omp_build_version
```

Comment:
> Can you suggest a workaround

I have one in mind is to create a pseduo build version based on system clang build version and its llvm version. However it has clearly potential harm. 

> is it hopeless?

This only means this approach is dead. We can still support clang-omp by treating it as a separate compiler.